nslsii/__init__.py

```diff
@@ -256,30 +259,37 @@ def configure_bluesky_logging(ipython):
     ----------
     ipython: InteractiveShell
         IPython InteractiveShell used to attach bluesky log handler to ipython
+    appdirs_appname: str
+        appname passed to appdirs.user_log_dir(); use the default for production,
+        set to something else for testing
 
     Returns
     -------
-    bluesky_log_file_path: str
+    bluesky_log_file_path: Path
         log file path
 
     """
     global bluesky_log_file_path
 
     if "BLUESKY_LOG_FILE" in os.environ:
+        bluesky_log_file_path = Path(os.environ["BLUESKY_LOG_FILE"])
         print(
             "bluesky log file configured from environment variable BLUESKY_LOG_FILE",
             file=sys.stderr,
         )
-        bluesky_log_file_path = os.environ["BLUESKY_LOG_FILE"]
     else:
+        bluesky_log_file_path = Path(
+            appdirs.user_log_dir(appname=appdirs_appname)
+        ) / Path("bluesky.log")
         print(
-            f"environment variable BLUESKY_LOG_FILE is not set, using default file path",
+            "environment variable BLUESKY_LOG_FILE is not set, using default file path",
             file=sys.stderr,
         )
-        bluesky_log_file_path = "/var/log/bluesky/bluesky.log"
+    if not bluesky_log_file_path.parent.exists():
```

Comment:
For deployment we should be setting the envs, this is so the out-of-the-box usage works (for example on my system I don't have write permissions to `/var/log` as me).  I think the default behavior should work rather than be tuned to our deployment details.

Reply:
We do point the ophyd and caproto loggers at the same file as the bluesky loggers. We are not doing that for databroker. Hmm. @danielballan should we point databroker loggers at the bluesky log file?

I would point LSDC logging at /var/log/LSDC, if it were up to me.

Reply:
Surfacing a comment publicly:

Yes, but it's not urgent because databroker currently does little to no logging.